storage/src/key_value/agent/mod.rs

```diff
@@ -41,18 +42,23 @@ use crate::{
     SerialTransaction,
     TransactionLocation,
     VMRecord,
+    Validator,
 };
 
 mod block;
 mod block_commit;
 
 pub(super) struct Agent<S: KeyValueStorage + 'static> {
-    inner: S,
+    inner: Option<S>, // the option is only for the purposes of validation, which requires ownership
```

Comment:
can we just decompose the agent? an agent with no inner just panics anyways -- and this makes the code less clean.